boot/freeldr/freeldr/ntldr/wlregistry.c

```diff
@@ -593,21 +609,112 @@ WinLdrScanRegistry(IN OUT PLIST_ENTRY BootDriverListHead,
                 rc = RegQueryValue(hDriverKey, L"Group", NULL, (PUCHAR)DriverGroup, &DriverGroupSize);
                 //TRACE_CH(REACTOS, "  Group: '%S'  \n", DriverGroup);
 
-                /* Make sure it should be started */
-                if ((StartValue == 0) &&
-                    (TagValue == OrderList[TagIndex]) &&
-                    (_wcsicmp(DriverGroup, GroupName) == 0))
+                /* Read the error control value */
+                ValueSize = sizeof(ULONG);
+                rc = RegQueryValue(hDriverKey, L"ErrorControl", &ValueType, (PUCHAR)&ErrorControl, &ValueSize);
+                if (rc != ERROR_SUCCESS) ErrorControl = 0;
+
+                /* Get the Driver's Location */
+                ValueSize = sizeof(TempImagePath);
+                rc = RegQueryValue(hDriverKey, L"ImagePath", NULL, (PUCHAR)TempImagePath, &ValueSize);
+
+                if (rc != ERROR_SUCCESS)
+                {
+                    ValueSize = 0;
+                }
+
+                ServiceNameLength = (wcslen(ServiceName) * sizeof(WCHAR)) + sizeof(WCHAR); // include NULL
```

Comment:
This function is becoming a mess. Please factor out the new stuff. Feel free to also factor out old stuff, but if you do so, do it in a separate commit for easier review. Also this is from a commit named "Speed up WinLdrScanRegistry", is this really a required change, or just a performance improvement? If it is the second, please open a separate PR for it and try to make it easier to review by splitting it into commits with smaller changes, that can actually be reviewed.

Reply:
Crikey, you've still not accepted this PR after a year? Yes, this was a required change - without it it would take quite literally half an hour to boot.

I'm not interested in making any changes to this PR, given I wrote it over a year ago... I won't be offended if you reject it, cannibalize it, whatever.